target/linux/ramips/image/rt3883.mk

```diff
@@ -1,15 +1,15 @@
 #
 # RT3662/RT3883 Profiles
 #
-define Build/mkrtn56uimg
-	$(STAGING_DIR_HOST)/bin/mkrtn56uimg $(1) $@
+define Build/asus-factory
+	$(STAGING_DIR_HOST)/bin/mkasusfw $(1) $(2) $@
```

Comment:
I don't think $(2) will work, as I understand it, $(1) contains all argument.. Look here for example:
https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/image/Makefile#L104

So, you won't need the $(2) at all if the $@ is not depending on it's position (haven't checked that).

Reply:
I've add a new arg for "asus-factory" tool so $(1) is a first arg, $(2) - second arg and $@ is the file name. Please correct me if I am wrong here...

Reply:
But we are not using $(call ...) to call the function, so from my experience (though I do not find it in the docs right now) $(1) is the list of arguments, and you have to use
```
  $(eval firstarg=$(word 1,$(1)))
  $(eval secondarg=$(word 2,$(1)))
```
if you want the individual value. However, in your case this is obviously irrelevant and it would be fine to do just:
`$(STAGING_DIR_HOST)/bin/mkasusfw $(1) $@`

A nice example of what I mean is iodata-factory here: https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/image/mt7621.mk#L51

Why is it still working for you? Because $(2) is just empty ...

Reply:
Thank you, got your point, fixed it